From 2fc7dbed4298b4ddcec5d3092b1ab39fcf884d71 Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Thu, 2 Oct 2025 14:59:19 +0200
Subject: [PATCH 406/553] Mark QWinRegistryKey as security-critical

It's a data parser, even though the parsing is generally rather
trivial, but it's also low-level, with unsecure C APIs being
called. The data is external, because the Windows registry is open
like the file system is. So if parsing from the file system is
critical (cf. QSettings), then so is parsing from the registry.

The header only contains trivial implementation, so mark it as
significant.

Task-number: QTBUG-135190
Pick-to: 6.8
Change-Id: I44ad39ec12460e862f97911bc68166cb2f7122cf
Reviewed-by: Edward Welbourne <edward.welbourne@qt.io>
(cherry picked from commit 5d746ad62b99152aae6ebcf2cf0b606b4d4982d5)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/kernel/qwinregistry.cpp | 1 +
 src/corelib/kernel/qwinregistry_p.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/corelib/kernel/qwinregistry.cpp b/src/corelib/kernel/qwinregistry.cpp
index 9740edbf299..fb315cacb7e 100644
--- a/src/corelib/kernel/qwinregistry.cpp
+++ b/src/corelib/kernel/qwinregistry.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2019 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:critical reason:data-parser
 
 #define UMDF_USING_NTSTATUS // Avoid ntstatus redefinitions
 
diff --git a/src/corelib/kernel/qwinregistry_p.h b/src/corelib/kernel/qwinregistry_p.h
index 5643a35363b..e8b749a5653 100644
--- a/src/corelib/kernel/qwinregistry_p.h
+++ b/src/corelib/kernel/qwinregistry_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2019 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:trivial-impl-only
 
 #ifndef QWINREGISTRY_H
 #define QWINREGISTRY_H
-- 
2.51.2

